### PR TITLE
Migrate makefiles and CI/CD

### PR DESCRIPTION
**Pull request by inteon**
ref: https://github.com/cert-manager/makefile-modules/issues/3

### CODE REVIEW

File: make/_shared/cert-manager/01_mod.mk [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cert_manager_crds := $(bin_dir)/scratch/cert-manager-crds.yaml
+$(cert_manager_crds): | $(bin_dir)/scratch
+	curl -sSLo $@ https://github.com/cert-manager/cert-manager/releases/download/$(cert_manager_version)/cert-manager.crds.yaml
```
Comment (SgtCoDFish) [inline make/_shared/cert-manager/01_mod.mk:17]: suggestion (not a blocker for this PR but maybe something wider): If we don't include a version number in the file we create, it won't be recreated when we bump `cert_manager_version`.

Could we do something like this?

```suggestion
cert_manager_crds := $(bin_dir)/scratch/cert-manager-crds-$(cert_manager_version).yaml
$(cert_manager_crds): | $(bin_dir)/scratch
	curl -sSLo $@ https://github.com/cert-manager/cert-manager/releases/download/$(cert_manager_version)/cert-manager.crds.yaml
```

Of course this means that the location is no longer constant but we can always pass the value of `cert_manager_crds` as a variable, or create a new `$(bin_dir)/scratch/cert-manager-crds.yaml` target which depends on `$(bin_dir)/scratch/cert-manager-crds-$(cert_manager_version).yaml`

File: make/_shared/tools/00_mod.mk [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+# https://github.com/helm/helm/releases
+TOOLS += helm=v3.13.3
+# https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl
+TOOLS += kubectl=v1.29.0
+# https://github.com/kubernetes-sigs/kind/releases
+TOOLS += kind=v0.20.0
+# https://www.vaultproject.io/downloads
+TOOLS += vault=1.15.4
+# https://github.com/Azure/azure-workload-identity/releases
+TOOLS += azwi=v1.2.0
```
Comment (SgtCoDFish) [inline make/_shared/tools/00_mod.mk:48]: question: Where's azwi coming from? I can't see it as currently being a dependency of istio-csr? Same question for vault I think and maybe others.

File: deploy/charts/istio-csr/templates/deployment.yaml [hunk 3]
```diff
@@ -45,7 +45,6 @@ spec:
             path: {{ .Values.app.readinessProbe.path }}
           initialDelaySeconds: 3
           periodSeconds: 7
-        command: ["cert-manager-istio-csr"]
```
Comment (SgtCoDFish) [inline deploy/charts/istio-csr/templates/deployment.yaml:48]: question: Why this change? Does the entrypoint change in the built container?

File: test/carotation/workloads/httpbin.yaml [hunk 4]
```diff
@@ -63,7 +63,7 @@ spec:
     spec:
       serviceAccountName: httpbin
       containers:
-      - image: quay.io/joshvanl_jetstack/httpbin:latest
```
Comment (SgtCoDFish) [inline test/carotation/workloads/httpbin.yaml:66]: comment: Yikes, good catch! I'd support moving this out of this PR and merging this immediately, but I'm also fine with including it here

**Review by SgtCoDFish — COMMENTED** [review]
Got a few questions: nothing major, but I'd like to understand more! I don't see any real blockers though.

File: make/_shared/tools/00_mod.mk (hunk 2 above)
Comment (inteon) [reply make/_shared/tools/00_mod.mk:48]: This library is part of the make-file modules, so it shouldn't be used in this project specifically.
That being said, it is also not used in approver-policy (the only other library that uses makefile-modules today), so we actually don't really need it.

File: make/_shared/cert-manager/01_mod.mk (hunk 1 above)
Comment (inteon) [reply make/_shared/cert-manager/01_mod.mk:17]: good catch, I'll change this. This is part of https://github.com/cert-manager/makefile-modules, so the fix will also improve approver-policy.

File: deploy/charts/istio-csr/templates/deployment.yaml (hunk 3 above)
Comment (inteon) [reply deploy/charts/istio-csr/templates/deployment.yaml:48]: Yes, I think the entrypoint changes.

File: make/_shared/cert-manager/01_mod.mk (hunk 1 above)
Comment (inteon) [reply make/_shared/cert-manager/01_mod.mk:17]: @SgtCoDFish https://github.com/cert-manager/makefile-modules/pull/17

File: test/carotation/workloads/httpbin.yaml (hunk 4 above)
Comment (inteon) [reply test/carotation/workloads/httpbin.yaml:66]: https://github.com/cert-manager/istio-csr/pull/226

Comment (inteon) [reply test/carotation/workloads/httpbin.yaml]: Will have to merge this PR first.

File: make/_shared/cert-manager/01_mod.mk (hunk 1 above)
Comment (inteon) [reply make/_shared/cert-manager/01_mod.mk:17]: We can merge that PR after this PR (or before).

File: test/carotation/workloads/httpbin.yaml (hunk 4 above)
Comment (SgtCoDFish) [reply test/carotation/workloads/httpbin.yaml]: Fair enough!

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

This is clearly reasonable! Thanks for this, I know it's tonnes of work!
